pkg/pillar/cmd/zedagent/parseconfig.go

```diff
@@ -1303,6 +1303,20 @@ func parseDeviceIoListConfig(getconfigCtx *getconfigContext,
 					"key: %s, value: %s", key, value)
 			}
 		}
+
+		if ioDevicePtr.Vflist != nil && ioDevicePtr.Vflist.VfCount > 0 {
+			port.Vfs.Count = uint8(ioDevicePtr.Vflist.VfCount)
+			port.Vfs.Data = make([]sriov.EthVF, ioDevicePtr.Vflist.VfCount)
```

Comment:
What if `port.Vfs.Data = make([]sriov.EthVF, min(port.Vfs.Count, len(ioDevicePtr.Vflist.Data)) ` ?
Just to be sure we don't crash in the loop below if Vfs.Count < len(ioDevicePtr.Vflist.Data).

Reply:
And to iterate over the `port.Vfs.Data` (of course assignment in the loop has to be changed accordingly).

Reply:
Yeah... I wasn't sure how "smart" to make this, but not crashing seems like a good goal.

There is also the possibility of if `len(ioDevicePtr.Vflist.Data) < ioDevicePtr.Vflist.Count`, we generate the unspecified ones but I figured I would send for review and get feedback.

Reply:
We even can just specify `VfList.Count` without specifying any of `VfList.Data`, i.e. we are creating virtual functions specified in vfList.Count and modify ones which are specified in VfList.Data linking it by ID. I.e. I want to create 5 virtual functions and vf3 should have specific parameters AA and vf4 BB

Reply:
On second thought, I think we should just ignore the entire PhysicalIO if there is a mismatch. I think things will be wrong if this isn't set up correctly anyways.

Reply:
I'd actually suggest to iterate over ioDevicePtr.VfList.Data and add necessary Vfs to it. Add it to code below which applies VLANs

Reply:
Oops, I missed this comment. ok, sgtm.

Reply:
Done.